### PR TITLE
Posdao test setup compatibility

### PR DESCRIPTION
**Pull request by LukaszRozmej**
1. Fixed random contract - wrong parameter
2. Added parity_enode JSON RPC
3. Added unlocking multiple accounts on startup based on password or password file
4. Added support for paths that are relative to current directory and not based on application directory

### CODE REVIEW

File: src/Nethermind/Nethermind.Config/ConfigSourceHelper.cs [hunk 1]
```diff
@@ -51,7 +51,13 @@ public static object ParseValue(Type valueType, string valueString)
                     var i = 0;
                     foreach (var valueItem in valueItems)
                     {
-                        var itemValue = GetValue(itemType, valueItem);
+                        string item = valueItem;
+                        if (valueItem.StartsWith('"') && valueItem.EndsWith('"'))
+                        {
+                            item = valueItem.Substring(1, valueItem.Length - 2);
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Config/ConfigSourceHelper.cs:57]: Trim("\"")?

Comment (tkstanczak) [reply src/Nethermind/Nethermind.Config/ConfigSourceHelper.cs:57]: Sure it does not break anything?

File: src/Nethermind/Nethermind.Runner/Ethereum/Steps/SetupKeyStore.cs [hunk 2]
```diff
@@ -65,13 +72,55 @@ public async Task Execute(CancellationToken cancellationToken)
...
                 _context.OriginalSignerKey = nodeKeyManager.LoadSignerKey();
                 _context.Enode = new Enode(_context.NodeKey.PublicKey, IPAddress.Parse(networkConfig.ExternalIp), networkConfig.P2PPort);
                 
                 _context.LogManager.SetGlobalVariable("enode", _context.Enode.ToString());
-            });
+            }, cancellationToken);
+        }
+
+        private void UnlockAccounts(IKeyStoreConfig config, IWallet wallet)
+        {
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Runner/Ethereum/Steps/SetupKeyStore.cs:87]: use IFileSystem and test

File: src/Nethermind/Nethermind.Config/ConfigSourceHelper.cs (hunk 1 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Config/ConfigSourceHelper.cs:57]: 1. It does not break anything, we didn't have string[] in configs yet.
2. Trim(""") would trim multiple characters, we just want to trim one from start and end if present.

File: src/Nethermind/Nethermind.Blockchain/Producers/BaseBlockProducer.cs [hunk 3]
```diff
@@ -161,7 +161,8 @@ protected virtual Block PrepareBlock(BlockHeader parent)
                 UInt256.Max(parent.Timestamp + 1, _timestamper.EpochSeconds),
                 Encoding.UTF8.GetBytes("Nethermind"))
             {
-                TotalDifficulty = parent.TotalDifficulty + difficulty
+                TotalDifficulty = parent.TotalDifficulty + difficulty,
+                Author = _sealer.Address
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Blockchain/Producers/BaseBlockProducer.cs:165]: Do we need Block.Author/Beneficiary as separate properties? One wouldn't suffice?

**Pull request merged**